xray/grpc_test.go

```diff
@@ -282,6 +287,28 @@ func TestUnaryServerInterceptor(t *testing.T) {
 			assert.Equal(t, header.Unknown, respTraceHeader.SamplingDecision)
 		})
 	}
+
+	// test that the interceptor by default will name the segment by the gRPC service name
+	t.Run("default namer", func(t *testing.T) {
+		ctx, td := NewTestDaemon()
+		defer td.Close()
+
+		lis := newGrpcServer(
+			t,
+			grpc_middleware.WithUnaryServerChain(
+				UnaryServerInterceptor(
+					ServerInterceptorWithContext(ctx),
+				),
+			),
+		)
+		client, closeFunc := newGrpcClient(context.Background(), t, lis)
+		defer closeFunc()
+		_, err := client.Ping(context.Background(), &pb.PingRequest{Value: "something", SleepTimeMs: 9999})
+		assert.NoError(t, err)
+		segment, err := td.Recv()
+		assert.NoError(t, err)
+		assert.Equal(t, "bufnet/mwitkow.testproto.TestService/Ping", segment.Name)
```

Comment:
Oh sorry also this should only be the service name by default, definitely not method and we don't need host in the default (a gRPC service name is semantically similar to a segment name, like a host would be on HTTP). So it should just be `mwitkow.testproto.TestService`

Reply:
Okay. Fixed